internal/authprocessor/authprocessor.go

```diff
@@ -0,0 +1,110 @@
+// Copyright 2024 go-dataspace
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package authprocessor checks if there's an authorization header, and then processes it into a
+// path prefix and injects it into the context.
+// It will construct an auth header like "foo;bar;baz" to "foo/bar/baz", which is used as a
```

Comment:
Please add a small description on the auth header format and why it is this way. Specifically saying something like "This is really not any kind of auth, but rather just a path description.".